src/plonk_util.rs

```diff
@@ -199,15 +209,18 @@ pub fn permutation_polynomial<F: Field>(
     gamma: F,
 ) -> Vec<F> {
     let mut plonk_z_points = vec![F::ONE];
+    let k_is = (0..NUM_ROUTED_WIRES)
+        .map(|j| get_subgroup_shift::<F>(j))
+        .collect::<Vec<_>>();
     for i in 1..degree {
-        let x = subgroup[i];
+        let x = subgroup[i - 1];
```

Comment:
I think the recursive `S_id` / `S_sigma` code in `verify_assumptions` may need to be tweaked to match this.

Reply:
After further inspection, I'm not totally sure changes are needed to the `verify_assumptions` function, since all the changes I made are taken care of in the openings of the `z` and `sigma` polynomials.